rust/parser/src/parser.rs

```diff
@@ -206,16 +206,4 @@ impl<'alloc> Parser<'alloc> {
         debug_assert_eq!(self.state_stack.len(), self.node_stack.len() + 1);
         state
     }
-
-    /// Return true if self.close() would succeed.
-    pub fn can_close(&self) -> bool {
-        // Easy case: no error, parsing just succeeds.
-        if self.can_accept_terminal(TerminalId::End) {
-            true
-        } else {
-            // Hard case: maybe error-handling would succeed?  BUG: Need
-            // simulator to simulate reduce_all; for now just give up
-            false
-        }
-    }
```

Comment:
https://github.com/mozilla-spidermonkey/jsparagus/pull/106 touches this,
so please revert the change for now.